src/main/java/edu/kit/compiler/optimizations/unrolling/LoopUnrollingOptimization.java

```diff
@@ -0,0 +1,170 @@
+package edu.kit.compiler.optimizations.unrolling;
+
+import static firm.bindings.binding_irgraph.ir_graph_properties_t.IR_GRAPH_PROPERTIES_NONE;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import edu.kit.compiler.optimizations.Optimization;
+import edu.kit.compiler.optimizations.OptimizationState;
+import edu.kit.compiler.optimizations.Util;
+import edu.kit.compiler.optimizations.unrolling.LoopAnalysis.Loop;
+import edu.kit.compiler.optimizations.unrolling.LoopAnalysis.LoopTree;
+import edu.kit.compiler.optimizations.unrolling.LoopVariableAnalysis.FixedIterationLoop;
+import edu.kit.compiler.transform.JFirmSingleton;
+import firm.Graph;
+import firm.Mode;
+import firm.Relation;
+import firm.TargetValue;
+import firm.bindings.binding_irgopt;
+import firm.nodes.Block;
+import firm.nodes.Node;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+
+public class LoopUnrollingOptimization implements Optimization.Local {
+
+    static {
+        JFirmSingleton.initializeFirmLinux();
+    }
```

Comment:
Personally, I would prefer a static method for `INT_MAX` instead of using such an initializer block